pkg/transport/managed_transport.go

```diff
@@ -412,7 +412,7 @@ func (mt *ManagedTransport) clearConn() {
 
 func (mt *ManagedTransport) updateStatus(isUp bool, tries int) (err error) {
 	if tries < 1 {
-		panic(fmt.Errorf("mt.updateStatus: invalid input: got tries=%d (want tries > 0)", tries))
```

Comment:
I feel a panic here is appropriate because it signals the developer that an invalid input is provided.
And the new error message is not clear and will be hard to track where the error came from.

Reply:
Probably it would be better if we just go for `uint` tries values here. We're calling this func passing `tries` as a raw number, not a variable, so no type conversions will be needed. And we won't need to decide on panic/error here

Reply:
Also, if we go with `uint` tries, we should change tries loop condition check inclusive down the code

Reply:
That's a good point. Thank you @Darkren

Reply:
I'm going to leave panic here.